openslides_backend/presenter/search_users_by_name_or_email.py

```diff
@@ -63,31 +64,41 @@ class SearchUsersByNameEmail(BasePresenter):
 
     def get_result(self) -> Any:
         self.check_permissions(self.data["permission_type"], self.data["permission_id"])
-        result: Dict[str, List[Dict[str, Union[str, int]]]] = {}
+        result: Dict[str, List[Dict[str, Union[str, int]]]] = defaultdict(list)
+        filter_bulk_tuples: Set[Tuple[str, str]] = set()
+        for search in self.data["search"]:
+            if username := search.get("username"):
+                filter_bulk_tuples.add(("username", username))
+            if email := search.get("email"):
+                filter_bulk_tuples.add(("email", email))
+        if len(filter_bulk_tuples) == 0:
+            return result
+        filter_bulk = Or(
+            *[FilterOperator(t[0], "~=", t[1]) for t in filter_bulk_tuples]
+        )
+        instances = self.datastore.filter(
+            "user",
+            filter_bulk,
+            ["id", "username", "first_name", "last_name", "email"],
+            lock_result=False,
+        )
+        userd: Dict[str, Set[int]] = defaultdict(set)
+        emaild: Dict[str, Set[int]] = defaultdict(set)
+        for instance in instances.values():
+            if username := instance["username"]:
+                userd[username.lower()].add(instance["id"])
+            if email := instance["email"]:
+                emaild[email.lower()].add(instance["id"])
         for search in self.data["search"]:
             username = search.get("username", "")
             email = search.get("email", "")
-            filter_: Filter
-            if username and email:
-                filter_ = Or(
-                    FilterOperator("username", "~=", username),
-                    FilterOperator("email", "~=", email),
-                )
-            elif username:
-                filter_ = FilterOperator("username", "~=", username)
-            elif email:
-                filter_ = FilterOperator("email", "~=", email)
-            else:
-                continue
-            instances = self.datastore.filter(
-                "user",
-                filter_,
-                ["id", "first_name", "last_name", "email"],
+            user_ids: Set[int] = userd.get(username.lower(), set()).union(
+                emaild.get(email.lower(), set())
```

Comment:
```suggestion
            user_ids: Set[int] = userd[username.lower()] | emaild[email.lower()]
```
you can use the index operator directly since they are defaultdicts